### PR TITLE
feat(appsync): support DEBUG and INFO logging levels for AppSync GraphQL APIs

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

N/A

### Reason for this change
AppSync now support `DEBUG` and `INFO` logging levels.

Announcement: [AWS AppSync enhances API monitoring with new DEBUG and INFO logging levels](https://aws.amazon.com/about-aws/whats-new/2024/09/aws-appsync-api-monitoring-debug-info-logging-levels/)

### Description of changes
* Added `INFO` and `DEBUG` to the FieldLogLevel enum.
* Added README, unit tests, and integration tests for FieldLogLevel as they were missing.

### Description of how you validated changes
Add unit tests and integ test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts [hunk 1]
```diff
@@ -186,15 +186,23 @@ export interface AuthorizationConfig {
  */
 export enum FieldLogLevel {
   /**
-   * No logging
+   * Resolver logging is disabled
```
Comment (mazyu36) [inline packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts:189]: Updated the descriptions for existing levels (NONE, ERROR, and ALL) to be more user-friendly, aligning them with the AWS Management Console.

image

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 2]
```diff
@@ -547,6 +547,10 @@ log data set to never expire. If you want to set a different expiration period,
 To obtain the GraphQL API's log group as a `logs.ILogGroup` use the `logGroup` property of the
 `GraphqlApi` construct.
 
+Also you can choose the log level by setting the `fieldLogLevel` property.
+
+For more infomertion, see [CloudWatch logs](https://docs.aws.amazon.com/en_us/appsync/latest/devguide/monitoring.html#cwl).
+
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-appsync/README.md:553]:
```suggestion
Also you can choose the log level by setting the `logConfig.fieldLogLevel` property.

For more information, see [CloudWatch logs](https://docs.aws.amazon.com/en_us/appsync/latest/devguide/monitoring.html#cwl).

To obtain the GraphQL API's log group as a `logs.ILogGroup` use the `logGroup` property of the
`GraphqlApi` construct.

```

Changes made to match the description before it:

```
AppSync automatically create a log group with the name `/aws/appsync/apis/<graphql_api_id>` upon deployment with
log data set to never expire. If you want to set a different expiration period, use the `logConfig.retention` property.
```

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 3]
```diff
@@ -558,7 +562,9 @@ new appsync.GraphqlApi(this, 'api', {
   authorizationConfig: {},
   name: 'myApi',
   definition: appsync.Definition.fromFile(path.join(__dirname, 'myApi.graphql')),
-  logConfig,
+  logConfig:{
+    fieldLogLevel: appsync.FieldLogLevel.INFO,
+  },
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-appsync/README.md:567]:
```suggestion
  logConfig: {
    fieldLogLevel: appsync.FieldLogLevel.INFO,
    retention: logs.RetentionDays.ONE_WEEK,
  },
```

And could you remove the line 557-560?

```ts
const logConfig: appsync.LogConfig = {
  retention: logs.RetentionDays.ONE_WEEK,
};
```

**Review by go-to-k — CHANGES_REQUESTED** [review]
Thanks for the PR. Only comments on doc.

File: packages/aws-cdk-lib/aws-appsync/README.md (hunk 2 above)
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-appsync/README.md:553]: This section could be moved below the sample code. You can decide if you prefer.

```
To obtain the GraphQL API's log group as a `logs.ILogGroup` use the `logGroup` property of the
`GraphqlApi` construct.
```

File: packages/aws-cdk-lib/aws-appsync/README.md (hunk 3 above)
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-appsync/README.md]: Thanks. 
Removed unnecessary lines.

**Comment by mazyu36** [comment]
@go-to-k 
Thank you for the review.
I’ve incorporated the comments.

**Review by go-to-k — APPROVED** [review]
LGTM!

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**